reolink/camera_api.py

```diff
@@ -402,6 +404,22 @@ async def get_stream_source(self):
 
         return stream_source
 
+    async def get_vod_source(self, filename: str):
+        """Return the vod source url."""
+        if not await self.login():
+            return
+
+        """
+        REOLink uses an odd encoding, if the camera provides a / in the filename it needs
+        to be encoded with %20
+        """
+        if self.protocol == DEFAULT_PROTOCOL:
+            stream_source = f"rtmp://{self._host}:{self._rtmp_port}/vod/{filename.replace('/', '%20')}?channel={self._channel}&stream=0&token={self._token}"
```

Comment:
Should the variable self._stream be used here? Please check async def get_stream_source. Stream number is different for RTSP.

Reply:
The vods (recordings) only come in the format it records in, and the api examples passed a 0 for stream, so I think it may have a different meaning or just be some sort of url requirement. I can verify this weekend, if it matters what is passed or if there is any error with passing stream.

Reply:
For a camera stream will always be 0. But for NVR, the channel is different for each camera. 
I find the replace of / for %20 (=space) very strange. Is this the only character that is translated?

Reply:
I agree about the %20, but I ran across an issue with one of my cameras, (510A) it returned Mp4Record/2021-03-01/RecM01_20210301_060852_061022_6D3C818_3254C5E.mp4 as a search result, and passing that back or through regular uri encoding would fail, so I asked REOLink and that was the response I got back, and it works. I haven't had any other issue as this works with the three cameras I have to test with. I have a fourth that I have not installed yet that I will test with as well, once I have it up and running,